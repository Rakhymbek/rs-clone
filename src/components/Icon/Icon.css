.Icon {
  cursor: pointer;

  transition: all 0.2s ease;
}

.Icon:hover {
}
