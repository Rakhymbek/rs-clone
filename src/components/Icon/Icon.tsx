import React from 'react';
import { FC } from 'react';
import { cn } from '@bem-react/classname';

import './Icon.css';

const cnIcon = cn('Icon');

export type IconProps = {
  type:
    | 'dislike'
    | 'like'
    | 'next'
    | 'note'
    | 'play'
    | 'prev'
    | 'repeat'
    | 'search'
    | 'shuffle'
    | 'sprite'
    | 'volume'
    | 'watch';
};

export const Icon: FC<IconProps> = ({ type }) => {
  return (
    <img src={`./icons/${type}.svg`} className={cnIcon()} alt={type}></img>
  );
};
