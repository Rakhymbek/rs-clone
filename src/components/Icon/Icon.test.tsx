import { render } from '@testing-library/react';
import { Icon } from './Icon';

describe('<Icon />', () => {
  it('should render successfully and match snapshot', () => {
    const { container } = render(<Icon type="like"></Icon>);

    expect(container).toMatchSnapshot();
  });
});
