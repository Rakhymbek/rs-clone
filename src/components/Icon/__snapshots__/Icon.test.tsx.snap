// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render successfully and match snapshot 1`] = `
<div>
  <img
    alt="like"
    class="Icon"
    src="./icons/like.svg"
  />
</div>
`;
